barter/src/engine/mod.rs

```diff
@@ -333,6 +333,7 @@ impl<Clock, InstrumentData, StrategyState, RiskState, ExecutionTxs, Strategy, Ri
         event: &AccountStreamEvent,
     ) -> UpdateFromAccountOutput<Strategy::OnDisconnect>
     where
+        InstrumentData: InstrumentDataState,
```

Comment:
Let's make this more general: 
`InstrumentData: for<'a> Processor<'a AccountEvent>`